core/modules/qmeta/QMetaMysql.h

```diff
@@ -235,6 +236,7 @@ class QMetaMysql : public QMeta {
 private:
 
     sql::SqlConnection _conn;
+    std::mutex _dbMutex;    ///< Synchronizes access to connection
```

Comment:
The comment should be better indeed. Not all access to _conn needs to be synchronized, only some operation (basically sending query and receiving data back) are not thread safe. But we end up synchronizing practically everything because we also want to synchronize transaction handling (no overlapping transactions on the same connection from different threads).